routers/web/repo/repo.go

```diff
@@ -579,20 +578,11 @@ func SearchRepo(ctx *context.Context) {
 	}
 
 	// collect the latest commit of each repo
-	repoIDsToLatestCommitSHAs := make(map[int64]string)
-	wg := sync.WaitGroup{}
-	wg.Add(len(repos))
+	// at most there are dozens of repos (limited by MaxResponseItems), so it's not a big problem at the moment
```

Comment:
(which is limited by RepoPagingNum)

Reply:
Are you sure?

The PageSize comes from `PageSize: convert.ToCorrectPageSize(ctx.FormInt("limit")),`,  and `ToCorrectPageSize` says:

```
else if size > setting.API.MaxResponseItems {
		size = setting.API.MaxResponseItems
	}
```

Reply:
I see, limit is RepoPagingNum but this condition overrides it